windows/drivers/uwb/simulator/UwbSimulatorDevice.cxx

```diff
@@ -209,11 +172,15 @@ UwbSimulatorDevice::OnFileCreate(WDFDEVICE device, WDFREQUEST request, WDFFILEOB
         TraceLoggingPointer(request, "Request"),
         TraceLoggingPointer(file, "File"));
 
-    auto uwbSimulatorFileBuffer = GetUwbSimulatorFile(file);
+    auto uwbSimulatorFileBuffer = GetUwbSimulatorDeviceFile(file);
     auto uwbSimulatorFile = new (uwbSimulatorFileBuffer) UwbSimulatorDeviceFile(file, this);
     auto uwbSimulatorFileStatus = uwbSimulatorFile->Initialize();
     if (uwbSimulatorFileStatus == STATUS_SUCCESS) {
         uwbSimulatorFile->RegisterHandler(m_ddiHandler);
+
+        std::unique_lock deviceFilesLockExclusive{ m_deviceFilesGate };
+        m_deviceFiles.push_back(uwbSimulatorFile);
+        DbgPrint("%p added file object %p\n", m_wdfDevice, file);
     } else {
         uwbSimulatorFile->~UwbSimulatorDeviceFile();

```

Comment:
I don't love this, but it seems like it might be necessary due to the placement new above? Is that necessary because of WDF reasons?

Reply:
Are you referring to manually invoking the destructor? If so, you're correct that it has to be invoked manually because it was constructed with a contiguous buffer from the WDF framework object, which doesn't have strong type-safety and so won't invoke any kind of destructor on it when it goes out of scope.

Reply:
Yes, that's what I was referring to, and your explanation reinforces my understanding. Thanks!